core/src/main/java/io/questdb/griffin/engine/functions/regex/RegexpReplaceStrFunctionFactory.java

```diff
@@ -0,0 +1,212 @@
+        private final StringBufferSink sinkB = new StringBufferSink();
+
+        public Func(Function value, Function pattern, int patternPos, Function replacement, int replacementPos, int maxLength) {
+            this.value = value;
+            this.pattern = pattern;
+            this.patternPos = patternPos;
+            this.replacement = replacement;
+            this.replacementPos = replacementPos;
+            this.maxLength = maxLength;
+        }
+
+        @Override
+        public Function getArg() {
+            return value;
+        }
+
+        @Override
+        public CharSequence getStr(Record rec) {
+            return getStr(rec, sink);
+        }
+
+        @Override
+        public CharSequence getStrB(Record rec) {
+            return getStr(rec, sinkB);
+        }
+
+        public CharSequence getStr(Record rec, StringBufferSink sink) {
+            CharSequence cs = value.getStr(rec);
+            if (cs == null) {
+                return null;
+            }
+
+            matcher.reset(cs);
+            sink.clear();
+
+            boolean result = matcher.find();
+            if (!result) {
+                sink.buffer.append(cs);
+            } else {
+                do {
+                    if (sink.length() > maxLength) {
+                        throw CairoException.nonCritical()
+                                .put("breached memory limit set for ").put(SIGNATURE)
+                                .put(" [maxLength=").put(maxLength).put(']');
+                    }
+                    matcher.appendReplacement(sink.buffer, replacementStr);
+                    result = matcher.find();
+                } while (result);
+                matcher.appendTail(sink.buffer);
+            }
+            return sink;
+        }
+
+        @Override
+        public boolean isConstant() {
+            return false;
+        }
+
+        @Override
+        public boolean isRuntimeConstant() {
+            return false;
+        }
+
+        @Override
+        public boolean isReadThreadSafe() {
+            return false;
+        }
+
+        @Override
+        public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) throws SqlException {
+            UnaryFunction.super.init(symbolTableSource, executionContext);
+            pattern.init(symbolTableSource, executionContext);
+            matcher = RegexUtils.createMatcher(pattern, patternPos);
+            replacement.init(symbolTableSource, executionContext);
+            CharSequence cs = replacement.getStr(null);
+            if (cs == null) {
+                throw SqlException.$(replacementPos, "NULL replacement");
+            }
+            replacementStr = cs.toString();
+        }
+    }
+
+    // TODO(puzpuzpuz):
+    //  Get rid of this class in favor of StringSink once we drop support for Java 8 where j.u.r.Matcher
+    //  has method overloads for StringBuilder.
```

Comment:
has **no** method overloads for StringBuilder? the word _no_ is missing?

Reply:
Indeed, it's a typo. Will fix.

Reply:
Done in f8022d255ec815f98350b81416b5b59d7bb7bf05